src/cache.ts

```diff
@@ -53,6 +53,48 @@ const wrapCacheEntry = (entry: CacheEntry): CachePayload => {
   };
 };
 
+/**
+ * Finds the corresponding key in the cache based on the provided key data.
+ * The keys inside the cache are in the format {prefix}::{client_id}::{audience}::{scope}.
+ * The first key in the cache that satisfies the following conditions is returned
+ *  - `prefix` is strict equal to Auth0's internally configured `keyPrefix`
+ *  - `client_id` is strict equal to the `cacheKeyData.client_id`
+ *  - `audience` is strict equal to the `cacheKeyData.audience`
+ *  - `scope` contains at least all the `cacheKeyData.scope` values
+ *  *
+ * @param cacheKeyData The provided cacheKeyData
+ * @param existingCacheKeys A list of existing cache keys
+ */
+const findExistingCacheKey = (
+  cacheKeyData: CacheKeyData,
+  existingCacheKeys: Array<string>
+) => {
+  const { client_id, audience, scope } = cacheKeyData;
+  return existingCacheKeys.find(key => {
+    const [
+      currentPrefix,
+      currentClientId,
+      currentAudience,
+      currentScopes
+    ] = key.split('::');
+    const currentScopesArr = currentScopes && currentScopes.split(' ');
+    const hasAllScopes =
+      currentScopes &&
+      scope
+        .split(' ')
+        .reduce(
+          (acc, current) => acc && currentScopesArr.includes(current),
+          true
+        );
+    return (
+      currentPrefix === keyPrefix &&
+      currentClientId === client_id &&
+      currentAudience === audience &&
+      hasAllScopes
+    );
+  });
+};
```

Comment:
Let's put some vertical whitespace in here between the assignments and the return block (similar to what we talked about in the past).

Reply:
Added some vertical whitespace